03week/towersOfHanoi.js

```diff
@@ -19,76 +19,119 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+// ======================================
+// move a single piece from the startStack to the endStack
+function movePiece(startStack, endStack) {
+  let playedPiece = stacks[startStack].pop(stacks[startStack].length - 1);
+  stacks[endStack].push(playedPiece);
 }
 
-function isLegal() {
-  // Your code here
-
+// console.log(movePiece);
+
+// ======================================
+// if the move is legal return true
+// if the move is not legal return false
+// a legal move is when there is at least 1 piece in the startStack AND:
+//    1. when the end stack is empty OR
+//    2. when the top piece in the start stack is smaller than the top piece in the end stack
+function isLegal(startStack, endStack) {
+  var s = stacks[startStack];
+  var e = stacks[endStack];
+  let topStartPiece = s[s.length - 1];
+  let topEndPiece = e[e.length - 1];
+
+  if (
+    s.length > 0 &&
+    (topStartPiece < topEndPiece || topEndPiece == undefined)
+  ) {
+    console.log("the move is legal");
+    movePiece(startStack, endStack);
+    return true;
+  } else {
+    console.log("the move is not legal");
+    return false;
+  }
 }
 
-function checkForWin() {
+// ======================================
+// if the player won, return true
+// if the player did not win, return false
+// a win is when all 4 blocks are in stack b or stack c
+function checkForWin(startStack, endStack) {
   // Your code here
-
+  if (stacks.b.length == 4 || stacks.c.length == 4) {
+    console.log("You Win!");
+    return true;
+  } else {
+    return false;
+  }
 }
 
+// ======================================
+// takes in the user input for startStack and endStack,
+// and digitally plays round
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  isLegal(startStack, endStack);
```

Comment:
instead of having movePeice being called from inside isLegal(), it should be called from here if the move is legal. That way the isLegal() function is just in charge of determining if the move is legal.